README
Astro::FITS::HdrTrans module
----------------------------

What is it?
-----------
Astro::FITS::HdrTrans and associated sub-classes are tools for translating
instrument-specific FITS headers into generically named headers.

Requirements
------------
The Astro::FITS::HdrTrans module has minimal requirements on other
modules; only Carp and Time::Piece are required. Instrument-specific
sub-classes of Astro::FITS::HdrTrans may require other modules to be
installed, depending on the transformations those sub-classes make.
As an example, the translation sub-class for IRIS2 (an imaging
spectrometer used at the Anglo-Australian Telescope) requires
Math::Trig in order to do zenith distance to airmass translations.

Adding Subclasses
-----------------
Adding subclasses to do translations for other instruments should
be fairly straight-forward. Detailed documentation can be found
in the Astro::FITS::HdrTrans::Base class.

Where can I get it?
-------------------
The package is available from CPAN <http://www.cpan.org/>.

Installation
------------
Installation is automated using the Module::Build library

    % perl Build.PL
    % ./Build
    % ./Build test
    % ./Build install

Authors
-------

The module was written by Brad Cavanagh <b.cavanagh@jach.hawaii.edu> and
Tim Jenness <t.jenness@jach.hawaii.edu> of the Joint Astronomy Centre
(JAC) in Hawaii, and Malcolm J. Currie <mjc@star.rl.ac.uk> of the Rutherford
Appleton Laboratory (RAL) and the JAC.

Copyright (C) 2007-2013 Science and Technology Facilities Council.
Copyright (C) 2003-2007 Particle Physics and Astronomy Research Council.
All Rights Reserved.

This program is free software; you can redistribute it and/or modify it under
the terms of the GNU General Public License as published by the Free Software
Foundation; either version 2 of the License, or (at your option) any later
version.

This program is distributed in the hope that it will be useful,but WITHOUT ANY
WARRANTY; without even the implied warranty of MERCHANTABILITY or FITNESS FOR A
PARTICULAR PURPOSE. See the GNU General Public License for more details.

You should have received a copy of the GNU General Public License along with
this program; if not, write to the Free Software Foundation, Inc., 59 Temple
Place,Suite 330, Boston, MA  02111-1307, USA

Maintainer
----------
The package is currently maintained by Tim Jenness <tjenness@cpan.org>.
Please use CPAN RT for bug reports and github pull requests for patches.
The source code repository is at

  git://github.com/Starlink/perl-Astro-FITS-HdrTrans.git


Changes
Revision history for Astro::FITS::HdrTrans

1.58 2015-03-08

    - SEEING for JCMT is now average of SEEINGST and SEEINGEN.
    - Add translation of SCAN_PAT header for JCMT.
    - Add translations for species/molecule and transition for ACSIS.

1.57 2015-03-06

    - Add support for more LCOGT instruments (Tim Lister).
    - Avoid NaN values for LCOGT (Tim Lister).

1.56 2014-07-16

    - For CGS4Old read INTTYPE if MODE is not present.
    - Add rotation for IRCAM.

1.55 2013-05-08

    - Updates to some LCOGT telescope translations (Tim Lister).
    - Added AMBIENT_TEMPERATURE for SCUBA-2.
    - No longer check OBSID for ACSIS.

1.54 2012-07-13

    - Update SCUBA-2 base coordinates estimate so that it can
      work with subheaders and missing start azel position.

1.53 2012-06-19

    - The ACSIS test should skip if Starlink::AST is missing.

1.52 2012-06-19

    - Add LCOGT instruments (thanks to Tim Lister).
    - Added OBSERVATION_ID_SUBSYSTEM
    - Many improvements to JCMT translations.
    - Astro::SLA is now only required for NIRI translations.

1.51 2011-01-04

    - Use Module::Build.
    - Minor fix for polarimeter recipe in ACSIS.

1.50 2010-08-26

    - Includes support for SCUBA-2 and many fixes for ACSIS.

0.05 2006-04-11

    - Major rewrite of internals.
    - Supports all UKIRT and JCMT instruments.

0.03 2005-03-29

    - Licence should have been GPL.

0.02 2003-09-16

    - Change licence in README to be same as Perl.
    - Add copyright statement.
    - Add requirement on Test::More and Time::Piece to Makefile.PL.

0.01 2003-09-15

    - Initial test release to CPAN.


Build.PL
#!perl

use strict;
use warnings;
use Module::Build;

# Copyright (C) 2010 Science and Technology Facilities Council.
# All Rights Reserved.

# This program is free software; you can redistribute it and/or modify
# it under the terms of the GNU General Public License as published by
# the Free Software Foundation; either version 2 of the License, or
# (at your option) any later version.

# This program is distributed in the hope that it will be useful,but
# WITHOUT ANY WARRANTY; without even the implied warranty of
# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the GNU
# General Public License for more details.

# You should have received a copy of the GNU General Public License
# along with this program; if not, write to the Free Software
# Foundation, Inc., 59 Temple Place,Suite 330, Boston, MA 02111-1307,
# USA

# Set it up.
my $build = Module::Build->new
  (
   module_name => 'Astro::FITS::HdrTrans',
   license  => 'gpl2',
   dist_abstract => 'Translate FITS headers to standardised form',
   dist_author => [
                   'Tim Jenness <t.jenness@jach.hawaii.edu>',
                   'Brad Cavanagh <b.cavanagh@jach.hawaii.edu>',
                   'Malcolm J. Currie <mjc@star.rl.ac.uk>',
                  ],
   dist_version => '1.58',
   script_files => [ qw/
                         translatehdr
                       / ],
   meta_merge => {
               resources =>  {
                              repository => "git://github.com/Starlink/perl-Astro-FITS-HdrTrans.git",
                             },
                },
   recommends => {
                  'Starlink::AST' => 1.02,
                  'Astro::SLA' => 0,
                 },
   requires => {
                'Astro::Coords' => 0,
                'Astro::FITS::Header' => 0,
                'Astro::Telescope' => 0,
                'DateTime' => 0,
                'DateTime::TimeZone' => 0,
                'Math::Trig' => 0,
                'Time::Piece' => 1.10,
                'Scalar::Util' => 0,
               },
   build_requires => {
                      'Test::More' => 0,
                      'Test::Number::Delta' => 0,
                      'Time::Piece' => 0,
                     },
   configure_requires => {
                          "Module::Build" => 0.30,
                         },
  );

$build->create_build_script;
